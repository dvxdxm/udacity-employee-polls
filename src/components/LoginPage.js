import { Fragment, useState } from "react";
import { connect } from "react-redux";
import { handleLoginUser } from "../actions/authedUser";
import { useNavigate } from "react-router-dom";

const LoginPage = ({ dispatch }) => {
  const navigate = useNavigate();
  const [user, setUser] = useState("");
  const [password, setPassword] = useState("");

  const handleSubmit = (e) => {
    e.preventDefault();

    dispatch(handleLoginUser({ user, password }));

    setUser("");
    setPassword("");

    navigate("/");
  };

  const handleUserInputChange = (e) => {
    const text = e.target.value;

    setUser(text);
  };

  const handlePasswordInputChange = (e) => {
    const text = e.target.value;

    setPassword(text);
  };

  const hasValidToSubmit = user && password;

  return (
    <Fragment>
      <h1 className="center" data-testid={"login-page-title"}>Employee Polls</h1>
      <div className="avartar-login">
        <img
          className="img-login"
          src="../images/avatar_login.jpg"
          alt="Avatar"
        />
      </div>
      <div>
        <h3 className="center">Login</h3>
        <form className="session-login" onSubmit={handleSubmit}>
          <div className="item">
            <span className="center">User</span>
            <div className="item-input">
              <input
                type="text"
                required
                maxLength={280}
                placeholder="User"
                value={user}
                onChange={handleUserInputChange}
              />
            </div>
          </div>
          <div className="item">
            <span className="center">Password</span>
            <div className="item-input">
              <input
                className="item-input"
                placeholder="Password"
                value={password}
                onChange={handlePasswordInputChange}
                maxLength={280}
                type="password"
              />
            </div>
          </div>
          <div className="login-btn">
            <button className="btn" type="submit" disabled={!hasValidToSubmit}>
              Submit
            </button>
          </div>
        </form>
      </div>
    </Fragment>
  );
};

export default connect()(LoginPage);
