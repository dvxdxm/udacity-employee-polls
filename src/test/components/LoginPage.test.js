import { render, screen } from "@testing-library/react";
import LoginPage from "../../components/LoginPage";
import { useNavigate } from "react-router-dom";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: jest.fn(),
}));

describe("LoginPage component", () => {
  const mockStore = configureMockStore();
  const store = mockStore({});
  
  it("should render success when access login to app", () => {
    const mockNavigate = jest.fn();
    useNavigate.mockReturnValue(mockNavigate);

    render(
      <Provider store={store}>
        <LoginPage />
      </Provider>
    );

    const title = screen.getByTestId("login-page-title");
    expect(title.innerHTML).toEqual("Employee Polls");

    const button = screen.getByText("Submit");
    expect(button.hasAttribute("disabled")).toBe(true);
  });
});
