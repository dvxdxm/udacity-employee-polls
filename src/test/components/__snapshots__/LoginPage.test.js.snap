// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage component will match snapshot 1`] = `
<body>
  <div>
    <h1
      class="center"
      data-testid="login-page-title"
    >
      Employee Polls
    </h1>
    <div
      class="avartar-login"
    >
      <img
        alt="Avatar"
        class="img-login"
        src="../images/avatar_login.jpg"
      />
    </div>
    <div>
      <h3
        class="center"
      >
        Login
      </h3>
      <form
        class="session-login"
      >
        <div
          class="item"
        >
          <span
            class="center"
          >
            User
          </span>
          <div
            class="item-input"
          >
            <input
              maxlength="280"
              placeholder="User"
              required=""
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="item"
        >
          <span
            class="center"
          >
            Password
          </span>
          <div
            class="item-input"
          >
            <input
              class="item-input"
              maxlength="280"
              placeholder="Password"
              type="password"
              value=""
            />
          </div>
        </div>
        <div
          class="login-btn"
        >
          <button
            class="btn"
            disabled=""
            type="submit"
          >
            Submit
          </button>
        </div>
      </form>
    </div>
  </div>
</body>
`;
